src/build/rollup-importmap-plugin.ts

```diff
@@ -0,0 +1,51 @@
+import { Plugin } from "rollup";
+import fs from "node:fs/promises";
+import path from "node:path";
+import { Flags } from "../types";
+import { getGenerator, JspmError } from "../utils";
+
+export const RollupImportmapPlugin = async (flags: Flags): Promise<Plugin> => {
+  const generator = await getGenerator({ ...flags, freeze: true });
+  await generator.install();
+
+  return {
+    name: "rollup-importmap-plugin",
+    resolveId: async (id: string, importer: string) => {
+      if (importer?.startsWith("http") && id?.startsWith(".")) {
+        const proxyPath = new URL(id, importer).toString();
+        return { id: proxyPath, external: false };
+      }
+
+      try {
+        const resolved = generator.importMap.resolve(id, importer);
+        return { id: resolved };
+      } catch (err) {
+        return { id, external: true };
+      }
+    },
+    load: async (id: string) => {
+      try {
+        const url = new URL(id);
+
+        if (url.protocol === "file:") {
+          /*
+            This is a hack and need to be replaced with a proper solution.
+          */
```

Comment:
Can you share why the extension adding is necessary?

Reply:
When users use imports without extensions like

```js
import { util  } from './utils'
```

The `load` fails to resolve this `./utils` I am not sure, if rollup allows to return these and let the other plugins like `rollup-plugin-node-resolve` handle the import.

Reply:
Now i am wondering should we even handle the imports without extensions 🤔

Reply:
In Node.js this only works when the package is loaded as CommonJS, not ES modules. And JSPM generator follows these rules, so I think we should do that.